tests/posix/src/semaphore.c

```diff
@@ -28,32 +28,46 @@ void test_posix_semaphore(void)
 	struct sched_param schedparam;
 	int schedpolicy = SCHED_FIFO;
 	int val, ret;
+	struct timespec abstime;
+
+	abstime.tv_nsec = 0;
+	abstime.tv_sec = 1;
 
 	schedparam.priority = 1;
-    ret = pthread_attr_init(&attr);
+
+	ret = pthread_attr_init(&attr);
 	if (ret != 0) {
-		zassert_false(pthread_attr_destroy(&attr),
-				"Unable to destroy pthread object attrib");
-		zassert_false(pthread_attr_init(&attr),
-				"Unable to create pthread object attrib");
+		zassert_equal(pthread_attr_destroy(&attr), 0,
+			      "Unable to destroy pthread object attrib");
+		zassert_equal(pthread_attr_init(&attr), 0,
+			      "Unable to create pthread object attrib");
 	}
 
 	pthread_attr_setstack(&attr, &stack, STACK_SIZE);
 	pthread_attr_setschedpolicy(&attr, schedpolicy);
 	pthread_attr_setschedparam(&attr, &schedparam);
 
 	zassert_equal(sem_init(&sema, 0, (CONFIG_SEM_VALUE_MAX + 1)), -1,
-			"value larger than %d\n", CONFIG_SEM_VALUE_MAX);
+		      "value larger than %d\n", CONFIG_SEM_VALUE_MAX);
 	zassert_equal(errno, EINVAL, NULL);
 
-	zassert_false(sem_init(&sema, 0, 0), "sem_init failed");
+	zassert_equal(sem_init(&sema, 0, 0), 0, "sem_init failed");
 
 	zassert_equal(sem_getvalue(&sema, &val), 0, NULL);
 	zassert_equal(val, 0, NULL);
 
 	ret = pthread_create(&newthread, &attr, child_func, NULL);
-	zassert_false(ret, "Thread creation failed");
+	zassert_equal(ret, 0, "Thread creation failed");
+
+	zassert_equal(sem_timedwait(&sema, &abstime), -1, NULL);
+	zassert_equal(errno, ETIMEDOUT, NULL);
```

Comment:
Add `sem_timedwait` with a successful return too, same for `sem_trywait` below

Reply:
For successful return, I would be required to change the current thread priority to preemptive. Is there any API in POSIX which changes the current thread priority?

Reply:
I'm not sure why you'd need to change the current thread priority here, but sure you can do it by calling `pthread_setschedparam` with the pthread_self after setting `schedparam.priority` appropriately.

Without that, you can move the call to `sched_yield` above, so the child thread does the `sem_post` and makes timed_wait successful. We can then play around with giving and taking multiple times like in the CMSIS semaphore test.

Reply:
Well, the reason why I asked function to change priority was, the timedwait was timing out without pending the thread. Later figured out that the issue was with the usage of timeout value specified in sem_timedwait. Anyways, your intention of code coverage is addressed now. Please have a look. Thanks!